benchmarks/bytecode/bytecode_profiler.py

```diff
@@ -0,0 +1,593 @@
+"""A bytecode performance profiler."""
+# TODO: Investigate specialising adaptive interpreter
+# TODO: `OpcodeEvent` could just store the instruction and make existing variables properties?
+# TODO: Split out into separate project
+# TODO: Make fun textual UI
```

Comment:
I would avoid TODOs. I would recommend the following approach for each task:

1. we definitely want it, and don't want to merge in this PR, add an issue
2. we might want it, not worth an issue nor a TODO

There's a risk of issue/todo fatigue, where it gets difficult to actually find the important things to tackle in the sea of things to do.